mongodb+srv connection strings are not working with latest quarkus version 3.19.1

Getting following exception. 

How to reproduce - just use the quickstart project and use mongodb+srv connection string https://github.com/quarkusio/quarkus-quickstarts/tree/main/mongodb-quickstart

2025-02-27 08:54:55,790 WARN  [io.net.res.dns.DefaultDnsServerAddressStreamProvider] (Quarkus Main Thread) Default DNS servers: [/[2001:4860:4860:0:0:0:0:8888]:53, /[2001:4860:4860:0:0:0:0:8844]:53] (Google Public DNS as a fallback)
2025-02-27 08:55:01,020 ERROR [io.qua.run.Application] (Quarkus Main Thread) Failed to start application: java.lang.RuntimeException: Failed to start quarkus
        at io.quarkus.runner.ApplicationImpl.doStart(Unknown Source)
        at io.quarkus.runtime.Application.start(Application.java:101)
        at io.quarkus.runtime.ApplicationLifecycleManager.run(ApplicationLifecycleManager.java:121)
        at io.quarkus.runtime.Quarkus.run(Quarkus.java:77)
        at io.quarkus.runtime.Quarkus.run(Quarkus.java:48)
        at io.quarkus.runtime.Quarkus.run(Quarkus.java:137)
        at io.quarkus.runner.GeneratedMain.main(Unknown Source)
        at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
        at java.base/java.lang.reflect.Method.invoke(Method.java:580)
        at io.quarkus.runner.bootstrap.StartupActionImpl$1.run(StartupActionImpl.java:116)
        at java.base/java.lang.Thread.run(Thread.java:1583)
Caused by: com.mongodb.MongoConfigurationException: Failed looking up TXT record for host XXXXXXXXXXXX.mongodb.net
        at com.mongodb.internal.dns.DefaultDnsResolver.resolveAdditionalQueryParametersFromTxtRecords(DefaultDnsResolver.java:148)
        at com.mongodb.ConnectionString.<init>(ConnectionString.java:458)
        at com.mongodb.ConnectionString.<init>(ConnectionString.java:343)
        at io.quarkus.mongodb.runtime.MongoClientRecorder.initializeDNSLookup(MongoClientRecorder.java:129)
        at io.quarkus.mongodb.runtime.MongoClientRecorder.performInitialization(MongoClientRecorder.java:118)
        at io.quarkus.runner.recorded.MongoClientProcessor$generateClientBeans405385475.deploy_0(Unknown Source)
        at io.quarkus.runner.recorded.MongoClientProcessor$generateClientBeans405385475.deploy(Unknown Source)
        ... 11 more
Caused by: com.mongodb.MongoConfigurationException: Unable to look up TXT record for host XXXXXXXXXXX.mongodb.net
        at io.quarkus.mongodb.runtime.dns.MongoDnsClient.resolveTxtRequest(MongoDnsClient.java:198)
        at io.quarkus.mongodb.runtime.dns.MongoDnsClient.getResourceRecordData(MongoDnsClient.java:108)
        at com.mongodb.internal.dns.DefaultDnsResolver.resolveAdditionalQueryParametersFromTxtRecords(DefaultDnsResolver.java:129)
        ... 17 more
Caused by: io.smallrye.mutiny.TimeoutException
        at io.smallrye.mutiny.operators.uni.UniBlockingAwait.await(UniBlockingAwait.java:64)
        at io.smallrye.mutiny.groups.UniAwait.atMost(UniAwait.java:65)
        at io.quarkus.mongodb.runtime.dns.MongoDnsClient.resolveTxtRequest(MongoDnsClient.java:196)

thanks @gsmet  for the reply. I am using a private link to connect to altas mongodb cluster. I just took the quarkus quickstart and used following configuration. I tried both dev and non dev mode of quarkus.

quarkus.mongodb.connection-string=mongodb+srv://username:XXpasswordXX@xxx-pl-0.xxx.mongodb.net/?retryWrites=true&w=majority&appName=myappname

quarkus.mongodb.dns.log-activity=true
quarkus.mongodb.dns.lookup-timeout=100


----------------------------LOGS -----------------

__  ____  __  _____   ___  __ ____  ______
 --/ __ \/ / / / _ | / _ \/ //_/ / / / __/
 -/ /_/ / /_/ / __ |/ , _/ ,< / /_/ /\ \
--\___\_\____/_/ |_/_/|_/_/|_|\____/___/
2025-02-27 14:27:26,930 WARN  [io.net.res.dns.DefaultDnsServerAddressStreamProvider] (main) Default DNS servers: [/[2001:4860:4860:0:0:0:0:8888]:53, /[2001:4860:4860:0:0:0:0:8844]:53] (Google Public DNS as a fallback)
2025-02-27 14:27:36,034 ERROR [io.qua.run.Application] (main) Failed to start application: java.lang.RuntimeException: Failed to start quarkus
        at io.quarkus.runner.ApplicationImpl.doStart(Unknown Source)
        at io.quarkus.runtime.Application.start(Application.java:101)
        at io.quarkus.runtime.ApplicationLifecycleManager.run(ApplicationLifecycleManager.java:121)
        at io.quarkus.runtime.Quarkus.run(Quarkus.java:77)
        at io.quarkus.runtime.Quarkus.run(Quarkus.java:48)
        at io.quarkus.runtime.Quarkus.run(Quarkus.java:137)
        at io.quarkus.runner.GeneratedMain.main(Unknown Source)
        at io.quarkus.bootstrap.runner.QuarkusEntryPoint.doRun(QuarkusEntryPoint.java:68)
        at io.quarkus.bootstrap.runner.QuarkusEntryPoint.main(QuarkusEntryPoint.java:36)
Caused by: com.mongodb.MongoConfigurationException: Failed looking up TXT record for host chubbio-icc-mordernization-pl-0.ryvw1.mongodb.net
        at com.mongodb.internal.dns.DefaultDnsResolver.resolveAdditionalQueryParametersFromTxtRecords(DefaultDnsResolver.java:148)
        at com.mongodb.ConnectionString.<init>(ConnectionString.java:458)
        at com.mongodb.ConnectionString.<init>(ConnectionString.java:343)
        at io.quarkus.mongodb.runtime.MongoClientRecorder.initializeDNSLookup(MongoClientRecorder.java:129)
        at io.quarkus.mongodb.runtime.MongoClientRecorder.performInitialization(MongoClientRecorder.java:118)
        at io.quarkus.runner.recorded.MongoClientProcessor$generateClientBeans405385475.deploy_0(Unknown Source)
        at io.quarkus.runner.recorded.MongoClientProcessor$generateClientBeans405385475.deploy(Unknown Source)
        ... 9 more
Caused by: com.mongodb.MongoConfigurationException: Unable to look up TXT record for host chubbio-icc-mordernization-pl-0.ryvw1.mongodb.net
        at io.quarkus.mongodb.runtime.dns.MongoDnsClient.resolveTxtRequest(MongoDnsClient.java:198)
        at io.quarkus.mongodb.runtime.dns.MongoDnsClient.getResourceRecordData(MongoDnsClient.java:108)
        at com.mongodb.internal.dns.DefaultDnsResolver.resolveAdditionalQueryParametersFromTxtRecords(DefaultDnsResolver.java:129)
        ... 15 more
Caused by: java.util.concurrent.CompletionException: java.net.SocketException: Network is unreachable: sendto
        at io.smallrye.mutiny.operators.uni.UniBlockingAwait.await(UniBlockingAwait.java:79)
        at io.smallrye.mutiny.groups.UniAwait.atMost(UniAwait.java:65)
        at io.quarkus.mongodb.runtime.dns.MongoDnsClient.resolveTxtRequest(MongoDnsClient.java:196)
        ... 17 more
Caused by: java.net.SocketException: Network is unreachable: sendto
        at java.base/sun.nio.ch.DatagramChannelImpl.send0(Native Method)
        at java.base/sun.nio.ch.DatagramChannelImpl.sendFromNativeBuffer(DatagramChannelImpl.java:1005)
        at java.base/sun.nio.ch.DatagramChannelImpl.send(DatagramChannelImpl.java:966)
        at java.base/sun.nio.ch.DatagramChannelImpl.send(DatagramChannelImpl.java:889)
        at io.netty.channel.socket.nio.NioDatagramChannel.doWriteMessage(NioDatagramChannel.java:297)
        at io.netty.channel.nio.AbstractNioMessageChannel.doWrite(AbstractNioMessageChannel.java:143)
        at io.netty.channel.AbstractChannel$AbstractUnsafe.flush0(AbstractChannel.java:929)
        at io.netty.channel.nio.AbstractNioChannel$AbstractNioUnsafe.flush0(AbstractNioChannel.java:359)
        at io.netty.channel.AbstractChannel$AbstractUnsafe.flush(AbstractChannel.java:893)
        at io.netty.channel.DefaultChannelPipeline$HeadContext.flush(DefaultChannelPipeline.java:1319)
        at io.netty.channel.AbstractChannelHandlerContext.invokeFlush0(AbstractChannelHandlerContext.java:935)
        at io.netty.channel.AbstractChannelHandlerContext.invokeFlush(AbstractChannelHandlerContext.java:921)
        at io.netty.channel.AbstractChannelHandlerContext.flush(AbstractChannelHandlerContext.java:907)
        at io.netty.handler.logging.LoggingHandler.flush(LoggingHandler.java:304)
        at io.netty.channel.AbstractChannelHandlerContext.invokeFlush0(AbstractChannelHandlerContext.java:937)
        at io.netty.channel.AbstractChannelHandlerContext.invokeFlush(AbstractChannelHandlerContext.java:921)
        at io.netty.channel.AbstractChannelHandlerContext.flush(AbstractChannelHandlerContext.java:907)
        at io.netty.channel.ChannelOutboundHandlerAdapter.flush(ChannelOutboundHandlerAdapter.java:125)
        at io.netty.channel.AbstractChannelHandlerContext.invokeFlush0(AbstractChannelHandlerContext.java:939)
        at io.netty.channel.AbstractChannelHandlerContext.invokeWriteAndFlush(AbstractChannelHandlerContext.java:957)
        at io.netty.channel.AbstractChannelHandlerContext.write(AbstractChannelHandlerContext.java:982)
        at io.netty.channel.AbstractChannelHandlerContext.writeAndFlush(AbstractChannelHandlerContext.java:950)
        at io.netty.channel.AbstractChannelHandlerContext.writeAndFlush(AbstractChannelHandlerContext.java:1000)
        at io.netty.channel.DefaultChannelPipeline.writeAndFlush(DefaultChannelPipeline.java:974)
        at io.netty.channel.AbstractChannel.writeAndFlush(AbstractChannel.java:305)
        at io.vertx.core.dns.impl.DnsClientImpl$Query.run(DnsClientImpl.java:366)
        at io.netty.util.concurrent.AbstractEventExecutor.runTask(AbstractEventExecutor.java:173)
        at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:166)
        at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:472)
        at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:569)
        at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:998)
        at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
        at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
        at java.base/java.lang.Thread.run(Thread.java:1583)
        Suppressed: java.lang.IllegalStateException: Retries exhausted: 3/3
                at io.smallrye.mutiny.helpers.ExponentialBackoff$1.lambda$apply$0(ExponentialBackoff.java:46)
                at io.smallrye.context.impl.wrappers.SlowContextualFunction.apply(SlowContextualFunction.java:21)
                at io.smallrye.mutiny.groups.MultiOnItem.lambda$transformToUni$6(MultiOnItem.java:268)
                at io.smallrye.mutiny.operators.multi.MultiConcatMapOp$MainSubscriber.onItem(MultiConcatMapOp.java:117)
                at io.smallrye.mutiny.subscription.MultiSubscriber.onNext(MultiSubscriber.java:61)
                at io.smallrye.mutiny.operators.multi.processors.UnicastProcessor.drainWithDownstream(UnicastProcessor.java:108)
                at io.smallrye.mutiny.operators.multi.processors.UnicastProcessor.drain(UnicastProcessor.java:139)
                at io.smallrye.mutiny.operators.multi.processors.UnicastProcessor.onNext(UnicastProcessor.java:205)
                at io.smallrye.mutiny.operators.multi.processors.SerializedProcessor.onNext(SerializedProcessor.java:104)
                at io.smallrye.mutiny.subscription.SerializedSubscriber.onItem(SerializedSubscriber.java:74)
                at io.smallrye.mutiny.subscription.MultiSubscriber.onNext(MultiSubscriber.java:61)
                at io.smallrye.mutiny.operators.multi.MultiRetryWhenOp$RetryWhenOperator.onFailure(MultiRetryWhenOp.java:127)
                at io.smallrye.mutiny.subscription.MultiSubscriber.onError(MultiSubscriber.java:73)
                at io.smallrye.mutiny.converters.uni.UniToMultiPublisher$UniToMultiSubscription.onFailure(UniToMultiPublisher.java:104)
                at io.smallrye.mutiny.vertx.AsyncResultUni.lambda$subscribe$1(AsyncResultUni.java:37)
                at io.vertx.core.impl.future.FutureImpl$4.onFailure(FutureImpl.java:188)
                at io.vertx.core.impl.future.FutureBase.emitFailure(FutureBase.java:81)
                at io.vertx.core.impl.future.FutureImpl.tryFail(FutureImpl.java:278)
                at io.vertx.core.impl.future.Mapping.onFailure(Mapping.java:45)
                at io.vertx.core.impl.future.FutureBase.emitFailure(FutureBase.java:81)
                at io.vertx.core.impl.future.FutureImpl.tryFail(FutureImpl.java:278)
                at io.vertx.core.Promise.fail(Promise.java:89)
                at io.vertx.core.impl.future.PromiseImpl.operationComplete(PromiseImpl.java:67)
                at io.netty.util.concurrent.DefaultPromise.notifyListener0(DefaultPromise.java:590)
                at io.netty.util.concurrent.DefaultPromise.notifyListenersNow(DefaultPromise.java:557)
                at io.netty.util.concurrent.DefaultPromise.notifyListeners(DefaultPromise.java:492)
                at io.netty.util.concurrent.DefaultPromise.setValue0(DefaultPromise.java:636)
                at io.netty.util.concurrent.DefaultPromise.setFailure0(DefaultPromise.java:629)
                at io.netty.util.concurrent.DefaultPromise.setFailure(DefaultPromise.java:110)
                at io.vertx.core.dns.impl.DnsClientImpl$Query.fail(DnsClientImpl.java:323)
                at io.vertx.core.impl.ContextImpl.emit(ContextImpl.java:342)
                at io.vertx.core.impl.ContextImpl.emit(ContextImpl.java:335)
                at io.vertx.core.dns.impl.DnsClientImpl$Query.lambda$run$2(DnsClientImpl.java:368)
                at io.netty.util.concurrent.DefaultPromise.notifyListener0(DefaultPromise.java:590)
                at io.netty.util.concurrent.DefaultPromise.notifyListenersNow(DefaultPromise.java:557)
                at io.netty.util.concurrent.DefaultPromise.notifyListeners(DefaultPromise.java:492)
                at io.netty.util.concurrent.DefaultPromise.addListener(DefaultPromise.java:185)
                at io.netty.channel.DefaultChannelPromise.addListener(DefaultChannelPromise.java:95)
                at io.netty.channel.DefaultChannelPromise.addListener(DefaultChannelPromise.java:30)
                ... 9 more
        Caused by: [CIRCULAR REFERENCE: java.net.SocketException: Network is unreachable: sendto]

> quarkus.mongodb.dns.lookup-timeout

Yes, i tried the same, it gives me the following error. Following is the config.

quarkus.mongodb.dns.lookup-timeout=200S
quarkus.mongodb.dns.log-activity=true

But if enable jndi using following, it starts working.
quarkus.naming.enable-jndi=true

__  ____  __  _____   ___  __ ____  ______
 --/ __ \/ / / / _ | / _ \/ //_/ / / / __/
 -/ /_/ / /_/ / __ |/ , _/ ,< / /_/ /\ \
--\___\_\____/_/ |_/_/|_/_/|_|\____/___/
2025-03-16 17:09:30,286 WARN  [io.net.res.dns.DefaultDnsServerAddressStreamProvider] (Quarkus Main Thread) Default DNS servers: [/[2001:4860:4860:0:0:0:0:8888]:53, /[2001:4860:4860:0:0:0:0:8844]:53] (Google Public DNS as a fallback)
2025-03-16 17:09:38,238 ERROR [io.qua.run.Application] (Quarkus Main Thread) Failed to start application: java.lang.RuntimeException: Failed to start quarkus
        at io.quarkus.runner.ApplicationImpl.doStart(Unknown Source)
        at io.quarkus.runtime.Application.start(Application.java:101)
        at io.quarkus.runtime.ApplicationLifecycleManager.run(ApplicationLifecycleManager.java:121)
        at io.quarkus.runtime.Quarkus.run(Quarkus.java:77)
        at io.quarkus.runtime.Quarkus.run(Quarkus.java:48)
        at io.quarkus.runtime.Quarkus.run(Quarkus.java:137)
        at io.quarkus.runner.GeneratedMain.main(Unknown Source)
        at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
        at java.base/java.lang.reflect.Method.invoke(Method.java:580)
        at io.quarkus.runner.bootstrap.StartupActionImpl$1.run(StartupActionImpl.java:116)
        at java.base/java.lang.Thread.run(Thread.java:1583)
Caused by: com.mongodb.MongoConfigurationException: Failed looking up TXT record for host privatelinkhostXXXXXXXX.mongodb.net
        at com.mongodb.internal.dns.DefaultDnsResolver.resolveAdditionalQueryParametersFromTxtRecords(DefaultDnsResolver.java:148)
        at com.mongodb.ConnectionString.<init>(ConnectionString.java:458)
        at com.mongodb.ConnectionString.<init>(ConnectionString.java:343)
        at io.quarkus.mongodb.runtime.MongoClientRecorder.initializeDNSLookup(MongoClientRecorder.java:129)
        at io.quarkus.mongodb.runtime.MongoClientRecorder.performInitialization(MongoClientRecorder.java:118)
        at io.quarkus.runner.recorded.MongoClientProcessor$generateClientBeans405385475.deploy_0(Unknown Source)
        at io.quarkus.runner.recorded.MongoClientProcessor$generateClientBeans405385475.deploy(Unknown Source)
        ... 11 more
Caused by: com.mongodb.MongoConfigurationException: Unable to look up TXT record for host privatelinkhostXXXXXXXX.mongodb.net
        at io.quarkus.mongodb.runtime.dns.MongoDnsClient.resolveTxtRequest(MongoDnsClient.java:198)
        at io.quarkus.mongodb.runtime.dns.MongoDnsClient.getResourceRecordData(MongoDnsClient.java:108)
        at com.mongodb.internal.dns.DefaultDnsResolver.resolveAdditionalQueryParametersFromTxtRecords(DefaultDnsResolver.java:129)
        ... 17 more
Caused by: java.util.concurrent.CompletionException: java.net.SocketException: Network is unreachable: sendto
        at io.smallrye.mutiny.operators.uni.UniBlockingAwait.await(UniBlockingAwait.java:79)
        at io.smallrye.mutiny.groups.UniAwait.atMost(UniAwait.java:65)
        at io.quarkus.mongodb.runtime.dns.MongoDnsClient.resolveTxtRequest(MongoDnsClient.java:196)
        ... 19 more
Caused by: java.net.SocketException: Network is unreachable: sendto
        at java.base/sun.nio.ch.DatagramChannelImpl.send0(Native Method)
        at java.base/sun.nio.ch.DatagramChannelImpl.sendFromNativeBuffer(DatagramChannelImpl.java:1005)
        at java.base/sun.nio.ch.DatagramChannelImpl.send(DatagramChannelImpl.java:966)
        at java.base/sun.nio.ch.DatagramChannelImpl.send(DatagramChannelImpl.java:889)
        at io.netty.channel.socket.nio.NioDatagramChannel.doWriteMessage(NioDatagramChannel.java:297)
        at io.netty.channel.nio.AbstractNioMessageChannel.doWrite(AbstractNioMessageChannel.java:143)
        at io.netty.channel.AbstractChannel$AbstractUnsafe.flush0(AbstractChannel.java:929)
        at io.netty.channel.nio.AbstractNioChannel$AbstractNioUnsafe.flush0(AbstractNioChannel.java:359)
        at io.netty.channel.AbstractChannel$AbstractUnsafe.flush(AbstractChannel.java:893)
        at io.netty.channel.DefaultChannelPipeline$HeadContext.flush(DefaultChannelPipeline.java:1319)
        at io.netty.channel.AbstractChannelHandlerContext.invokeFlush0(AbstractChannelHandlerContext.java:935)
        at io.netty.channel.AbstractChannelHandlerContext.invokeFlush(AbstractChannelHandlerContext.java:921)
        at io.netty.channel.AbstractChannelHandlerContext.flush(AbstractChannelHandlerContext.java:907)
        at io.netty.handler.logging.LoggingHandler.flush(LoggingHandler.java:304)
        at io.netty.channel.AbstractChannelHandlerContext.invokeFlush0(AbstractChannelHandlerContext.java:937)
        at io.netty.channel.AbstractChannelHandlerContext.invokeFlush(AbstractChannelHandlerContext.java:921)
        at io.netty.channel.AbstractChannelHandlerContext.flush(AbstractChannelHandlerContext.java:907)
        at io.netty.channel.ChannelOutboundHandlerAdapter.flush(ChannelOutboundHandlerAdapter.java:125)
        at io.netty.channel.AbstractChannelHandlerContext.invokeFlush0(AbstractChannelHandlerContext.java:939)
        at io.netty.channel.AbstractChannelHandlerContext.invokeWriteAndFlush(AbstractChannelHandlerContext.java:957)
        at io.netty.channel.AbstractChannelHandlerContext.write(AbstractChannelHandlerContext.java:982)
        at io.netty.channel.AbstractChannelHandlerContext.writeAndFlush(AbstractChannelHandlerContext.java:950)
        at io.netty.channel.AbstractChannelHandlerContext.writeAndFlush(AbstractChannelHandlerContext.java:1000)
        at io.netty.channel.DefaultChannelPipeline.writeAndFlush(DefaultChannelPipeline.java:974)
        at io.netty.channel.AbstractChannel.writeAndFlush(AbstractChannel.java:305)
        at io.vertx.core.dns.impl.DnsClientImpl$Query.run(DnsClientImpl.java:366)
        at io.netty.util.concurrent.AbstractEventExecutor.runTask(AbstractEventExecutor.java:173)
        at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:166)
        at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:472)
        at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:569)
        at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:998)
        at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
        at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
        ... 1 more
        Suppressed: java.lang.IllegalStateException: Retries exhausted: 3/3
                at io.smallrye.mutiny.helpers.ExponentialBackoff$1.lambda$apply$0(ExponentialBackoff.java:46)
                at io.smallrye.context.impl.wrappers.SlowContextualFunction.apply(SlowContextualFunction.java:21)
                at io.smallrye.mutiny.groups.MultiOnItem.lambda$transformToUni$6(MultiOnItem.java:268)
                at io.smallrye.mutiny.operators.multi.MultiConcatMapOp$MainSubscriber.onItem(MultiConcatMapOp.java:117)
                at io.smallrye.mutiny.subscription.MultiSubscriber.onNext(MultiSubscriber.java:61)
                at io.smallrye.mutiny.operators.multi.processors.UnicastProcessor.drainWithDownstream(UnicastProcessor.java:108)
                at io.smallrye.mutiny.operators.multi.processors.UnicastProcessor.drain(UnicastProcessor.java:139)
                at io.smallrye.mutiny.operators.multi.processors.UnicastProcessor.onNext(UnicastProcessor.java:205)
                at io.smallrye.mutiny.operators.multi.processors.SerializedProcessor.onNext(SerializedProcessor.java:104)
                at io.smallrye.mutiny.subscription.SerializedSubscriber.onItem(SerializedSubscriber.java:74)
                at io.smallrye.mutiny.subscription.MultiSubscriber.onNext(MultiSubscriber.java:61)
                at io.smallrye.mutiny.operators.multi.MultiRetryWhenOp$RetryWhenOperator.onFailure(MultiRetryWhenOp.java:127)
                at io.smallrye.mutiny.subscription.MultiSubscriber.onError(MultiSubscriber.java:73)
                at io.smallrye.mutiny.converters.uni.UniToMultiPublisher$UniToMultiSubscription.onFailure(UniToMultiPublisher.java:104)
                at io.smallrye.mutiny.vertx.AsyncResultUni.lambda$subscribe$1(AsyncResultUni.java:37)
                at io.vertx.core.impl.future.FutureImpl$4.onFailure(FutureImpl.java:188)
                at io.vertx.core.impl.future.FutureBase.emitFailure(FutureBase.java:81)
                at io.vertx.core.impl.future.FutureImpl.tryFail(FutureImpl.java:278)
                at io.vertx.core.impl.future.Mapping.onFailure(Mapping.java:45)
                at io.vertx.core.impl.future.FutureBase.emitFailure(FutureBase.java:81)
                at io.vertx.core.impl.future.FutureImpl.tryFail(FutureImpl.java:278)
                at io.vertx.core.Promise.fail(Promise.java:89)
                at io.vertx.core.impl.future.PromiseImpl.operationComplete(PromiseImpl.java:67)
                at io.netty.util.concurrent.DefaultPromise.notifyListener0(DefaultPromise.java:590)
                at io.netty.util.concurrent.DefaultPromise.notifyListenersNow(DefaultPromise.java:557)
                at io.netty.util.concurrent.DefaultPromise.notifyListeners(DefaultPromise.java:492)
                at io.netty.util.concurrense.setValue0(DefaultPromise.java:636)
                at io.netty.util.concurrent.DefaultPromise.setFailure0(DefaultPromise.java:629)
                at io.netty.util.concurrent.DefaultPromise.setFailure(DefaultPromise.java:110)
                at io.vertx.core.dns.impl.DnsClientImpl$Query.fail(DnsClientImpl.java:323)
                at io.vertx.core.impl.ContextImpl.emit(ContextImpl.java:342)
                at io.vertx.core.impl.ContextImpl.emit(ContextImpl.java:335)
                at io.vertx.core.dns.impl.DnsClientImpl$Query.lambda$run$2(DnsClientImpl.java:368)
                at io.netty.util.concurrent.DefaultPromise.notifyListener0(DefaultPromise.java:590)
                at io.netty.util.concurrent.DefaultPromise.notifyListenersNow(DefaultPromise.java:557)
                at io.netty.util.concurrent.DefaultPromise.notifyListeners(DefaultPromise.java:492)
                at io.netty.util.concurrent.DefaultPromise.addListener(DefaultPromise.java:185)
                at io.netty.channel.DefaultChannelPromise.addListener(DefaultChannelPromise.java:95)
                at io.netty.channel.DefaultChannelPromise.addListener(DefaultChannelPromise.java:30)
                ... 9 more
        Caused by: [CIRCULAR REFERENCE: java.net.SocketException: Network is unreachable: sendto]

--- a/extensions/mongodb-client/runtime/src/main/java/io/quarkus/mongodb/runtime/MongodbConfig.java
+++ b/extensions/mongodb-client/runtime/src/main/java/io/quarkus/mongodb/runtime/MongodbConfig.java
@@ -100,14 +100,12 @@ public interface MongodbConfig {
      */
     @Deprecated
     @WithName(NATIVE_DNS_SERVER_PORT)
-    @WithDefault("53")
     OptionalInt dnsServerPortInNativeMode();
 
     /**
      * This property configures the DNS server port.
      */
     @WithName(DNS_SERVER_PORT)
-    @WithDefault("53")
     OptionalInt dnsServerPort();
 
     /**

--- a/extensions/mongodb-client/runtime/src/main/java/io/quarkus/mongodb/runtime/dns/MongoDnsClient.java
+++ b/extensions/mongodb-client/runtime/src/main/java/io/quarkus/mongodb/runtime/dns/MongoDnsClient.java
@@ -57,6 +57,8 @@ public class MongoDnsClient implements DnsClient {
     private static final Map<String, List<SrvRecord>> SRV_CACHE = new ConcurrentHashMap<>();
     private static final Map<String, List<String>> TXT_CACHE = new ConcurrentHashMap<>();
 
+    private static final Logger log = Logger.getLogger(MongoDnsClient.class);
+
     MongoDnsClient(io.vertx.core.Vertx vertx) {
         Vertx mutinyVertx = new io.vertx.mutiny.core.Vertx(vertx);
 
@@ -74,12 +76,14 @@ public class MongoDnsClient implements DnsClient {
         DnsClientOptions dnsClientOptions = new DnsClientOptions()
                 .setLogActivity(activity);
         if (server != null) {
-            int port = config.getOptionalValue(DNS_SERVER_PORT, Integer.class)
-                    .orElse(53);
-            dnsClientOptions
-                    .setHost(server)
-                    .setPort(port);
+            dnsClientOptions.setHost(server);
+            if (config.getOptionalValue(DNS_SERVER_PORT, Integer.class).isPresent()) {
+                dnsClientOptions.setPort(config.getOptionalValue(DNS_SERVER_PORT, Integer.class).orElseThrow());
+            }
         }
+        dnsClientOptions.setQueryTimeout(config.getValue(DNS_LOOKUP_TIMEOUT, Duration.class).toMillis());
+
+        log.debugf("DNS client options: %s", dnsClientOptions.toJson());
         dnsClient = mutinyVertx.createDnsClient(dnsClientOptions);
     }
 
@@ -101,14 +105,12 @@ public class MongoDnsClient implements DnsClient {
 
     @Override
     public List<String> getResourceRecordData(String name, String type) throws DnsException {
-        switch (type) {
-            case "SRV":
-                return resolveSrvRequest(name);
-            case "TXT":
-                return resolveTxtRequest(name);
-            default:
-                throw new IllegalArgumentException("Unknown DNS record type: " + type);
-        }
+        log.debugf("Resolving DNS record for name: %s and type: %s", name, type);
+        return switch (type) {
+            case "SRV" -> resolveSrvRequest(name);
+            case "TXT" -> resolveTxtRequest(name);
+            default -> throw new IllegalArgumentException("Unknown DNS record type: " + type);
+        };
     }
 
     /*
@@ -152,13 +154,17 @@ public class MongoDnsClient implements DnsClient {
             if (srvRecords.isEmpty()) {
                 throw new MongoConfigurationException("No SRV records available for host " + srvHost);
             }
+            List<String> results = new ArrayList<>();
             for (SrvRecord srvRecord : srvRecords) {
                 String resolvedHost = srvRecord.target().endsWith(".")
                         ? srvRecord.target().substring(0, srvRecord.target().length() - 1)
                         : srvRecord.target();
 
-                hosts.add(format("%d %d %d %s", srvRecord.priority(), srvRecord.weight(), srvRecord.port(), resolvedHost));
+                var r = format("%d %d %d %s", srvRecord.priority(), srvRecord.weight(), srvRecord.port(), resolvedHost);
+                results.add(r);
             }
+            hosts.addAll(results);
+            log.debugf("Resolved SRV records for %s: %s", srvHost, String.join(", ", results));
         } catch (Throwable e) {
             throw new MongoConfigurationException("Unable to look up SRV record for host " + srvHost, e);
         }
@@ -190,6 +196,7 @@ public class MongoDnsClient implements DnsClient {
                     .invoke(new Consumer<>() {
                         @Override
                         public void accept(List<String> strings) {
+                            log.debugf("Resolved TXT records for %s: %s", host, strings);
                             TXT_CACHE.put(host, strings);
                         }
                     })

